api/src/services/checkouts/checkouts.js

```diff
@@ -1,24 +1,26 @@
-import { stripe } from 'src/lib/stripe'
 import { db } from 'src/lib/db'
+import { stripe } from 'src/lib/stripe'
 
 /**
- * @param {'payment' | 'subscription'} mode
- * @param {{ id: string, quantity: number }} cart
+ * @param {{
+ *   mode: 'payment' | 'subscription'
+ *   cart: Array<{ id: string, quantity: number }>
+ *   customerId?: string
+ * }}
+ *
+ * @returns {import('stripe').Stripe.Checkout.Session}
  */
-export const checkout = async ({ mode, cart, customerId }) => {
+export const createCheckoutSession = async ({ mode, cart, customerId }) => {
   // eslint-disable-next-line camelcase
   const line_items = cart.map((product) => ({
     price: product.id,
     quantity: product.quantity,
   }))
 
   return stripe.checkout.sessions.create({
-    success_url: `${
-      context.request?.headers?.referer ?? process.env.DOMAIN_URL
-    }success?sessionId={CHECKOUT_SESSION_ID}`,
-    cancel_url: `${
-      context.request?.headers?.referer ?? process.env.DOMAIN_URL
-    }failure`,
+    // See https://stripe.com/docs/payments/checkout/custom-success-page#modify-success-url.
+    success_url: `${context.event.headers.referer}success?sessionId={CHECKOUT_SESSION_ID}`,
+    cancel_url: `${context.event.headers.referer}failure`,
```

Comment:
`context.event.headers.referer` should work now; I didn't account for the fact that we were using Fastify before